specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/networkManagerEffectiveConfiguration.json

```diff
@@ -89,41 +89,286 @@
+            "ConnectivityConfiguration",
+            "SecurityRuleCollection"
           ],
           "x-ms-enum": {
-            "name": "ConfigType",
+            "name": "kind",
             "modelAsString": true
           },
-          "description": "Effective configuration."
-        },
-        "connectivityConfiguration": {
-          "$ref": "./networkManagerConnectivityConfiguration.json#/definitions/ConnectivityConfiguration",
-          "description": "Connectivity configuration object."
-        },
-        "securityAdminRule": {
-          "$ref": "./networkManagerSecurityConfiguration.json#/definitions/AdminRule",
-          "description": "Security admin rule object."
-        },
-        "securityUserRule": {
-          "$ref": "./networkManagerSecurityConfiguration.json#/definitions/UserRule",
-          "description": "Security user rule object."
+          "description": "Effective configuration Type."
         },
         "configurationGroups": {
           "type": "array",
           "items": {
-            "$ref": "./networkManagerGroup.json#/definitions/NetworkGroup"
+            "$ref": "#/definitions/ConfigurationGroup"
           },
           "description": "Effective configuration groups."
         }
       },
+      "required": [
+        "kind"
+      ],
+      "discriminator": "kind",
       "description": "The network manager effective configuration"
     },
+    "EffectiveConnectivityConfiguration": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/EffectiveConnectivityConfigurationProperties",
+          "description": "Properties of a network manager connectivity configuration"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/EffectiveConfiguration"
+        }
+      ],
+      "description": "The network manager connectivity configuration resource",
+      "x-ms-discriminator-value": "ConnectivityConfiguration"
+    },
+    "SecurityRuleCollection": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/RuleCollectionProperties",
+          "description": "The security rule collection properties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/EffectiveConfiguration"
+        }
+      ],
+      "description": "The network manager effective security rule collection",
+      "x-ms-discriminator-value": "SecurityRuleCollection"
+    },
+    "EffectiveConnectivityConfigurationProperties": {
+      "properties": {
+        "displayName": {
+          "type": "string",
+          "description": "A friendly name for the resource."
+        },
+        "description": {
+          "type": "string",
+          "description": "A description of the connectivity configuration."
+        },
+        "connectivityTopology": {
+          "type": "string",
+          "enum": [
+            "HubAndSpokeTopology",
```

Comment:
nit: remove Topology in enum values since the propertyName connectivityTopology already implies it